networkx/linalg/bethehessianmatrix.py

```diff
@@ -42,7 +41,14 @@ def bethe_hessian_matrix(G, r=None, nodelist=None):
     --------
     >>> k = [3, 2, 2, 1, 0]
     >>> G = nx.havel_hakimi_graph(k)
-    >>> H = nx.modularity_matrix(G)
+    >>> H = nx.bethe_hessian_matrix(G)
+    >>> H.todense()
+    matrix([[ 3.5625, -1.25  , -1.25  , -1.25  ,  0.    ],
+            [-1.25  ,  2.5625, -1.25  ,  0.    ,  0.    ],
+            [-1.25  , -1.25  ,  2.5625,  0.    ,  0.    ],
+            [-1.25  ,  0.    ,  0.    ,  1.5625,  0.    ],
+            [ 0.    ,  0.    ,  0.    ,  0.    ,  0.5625]])
```

Comment:
```suggestion
    >>> H.toarray()
    array([[ 3.5625, -1.25  , -1.25  , -1.25  ,  0.    ],
            [-1.25  ,  2.5625, -1.25  ,  0.    ,  0.    ],
            [-1.25  , -1.25  ,  2.5625,  0.    ,  0.    ],
            [-1.25  ,  0.    ,  0.    ,  1.5625,  0.    ],
            [ 0.    ,  0.    ,  0.    ,  0.    ,  0.5625]])
```
NetworkX is moving away from using `numpy.matrix` as much as possible, so using `toarray` here instead of `tomatrix` would be preferable.

Reply:
@rossbar thank you for the feedback, I was aware of the open PR. Since it is stated that (i) providing example and (ii) putting underscore after references remained unsolved (see below comment from that PR), I thought that I could work on these parts.

_Thanks for the updates to the Returns and References. The two remaining things that should be fixed to close https://github.com/networkx/networkx/issues/5192 are:_

_Replace the docstring Example with something relevant (e.g. replace modularity_matrix with bethe_hessian_matrix and show the output._
_In the locations where the references are linked, make sure that the ref is followed by an underscore, e.g. [2]_ per the [docstring standard](https://numpydoc.readthedocs.io/en/latest/format.html#references)_

I will certainly keep an eye on the open PRs for future contributions too :)